sklearn/compose/_column_transformer.py

```diff
@@ -1371,3 +1430,115 @@ def __call__(self, df):
         if self.pattern is not None:
             cols = cols[cols.str.contains(self.pattern, regex=True)]
         return cols.tolist()
+
+
+class _RemainderColsList(UserList):
```

Comment:
I think that we should change the representation of this class:

```
[('num',
  Pipeline(steps=[('imputer', SimpleImputer(strategy='median')),
                  ('scaler', StandardScaler())]),
  ['age', 'fare']),
 ('cat',
  Pipeline(steps=[('encoder', OneHotEncoder(handle_unknown='ignore')),
                  ('selector',
                   SelectPercentile(percentile=50,
                                    score_func=<function chi2 at 0x137e95870>))]),
  ['embarked', 'sex', 'pclass']),
 ('remainder', 'drop', _RemainderColsList([1, 4, 5, 6, 8, 10, 11, 12]))]
```

This is a bit uggly to see a private `_RemainderColsList`. We should show it as a list. This will be less surprising for the user.

Reply:
we needed to define `_repr_pretty_` to fix the display in the ipython console